docs/reference/settings.rst

```diff
@@ -6,6 +6,38 @@ Settings
 
 :ref:`Settings <topics-settings>` for scrapy-zyte-api.
 
+.. setting:: ZYTE_API_AUTO_FIELD_STATS
+
+ZYTE_API_AUTO_FIELD_STATS
+=========================
+
+Default: ``False``
+
+Enables stats that indicate which requested fields come directly from
+:ref:`zyte-api-extract`.
+
+If for any request no page object class is used to override
+:ref:`zyte-api-extract` fields for a given item type, the following stat is
+set:
+
+.. code-block:: python
+
+    "scrapy-zyte-api/auto_fields/<item class import path>": "<space-separated field list>"
```

Comment:
How would this stat work if there are several page objects for the same item type, but for different websites / paths?

Reply:
This stat only happens when there is no page object for a given website, and instead the item is directly provided by Zyte API.

If there is a page object, the other stat is used instead.

Reply:
I guess it works as a way to know if Zyte API Extraction was used. Is there any other stats to indicate that Zyte API Extraction was used without a Page Object?

Reply:
There’s stats for Zyte API fields used in requests, e.g.:

```
scrapy-zyte-api/request_args/browserHtml	2
scrapy-zyte-api/request_args/jobId	22
scrapy-zyte-api/request_args/product	20
scrapy-zyte-api/request_args/productNavigation	2
scrapy-zyte-api/request_args/url	22
```

Reply:
> I guess it works as a way to know if Zyte API Extraction was used. Is there any other stats to indicate that Zyte API Extraction was used without a Page Object?

What does it mean exactly, by the way? Is it "callback has an item dependency, and to resolve this dependency scrapy-zyte-api provider was used directly, witthout page objects involved"?  I wonder if we can add a related stat to scrapy-poet (not here).

Reply:
If the goal is to see which items are not overwritten, it sounds easier to not have the field lists in the stat value, and have simple item:value counters instead.

Reply:
For my side of functionality (find overwritten fields) item counts are irrelevant, but I could see some use for this when digging deeper into multi-item or multi-domain jobs.

Reply:
@asadurski My main question for you would be:
- Whether or not we should log in stats something at all in cases where no override is used. I imagine it might be useful in multi-domain spiders, to tell when both overrides and non-overrides were used.
- If we do log it in stats, if there is any point in listing its fields (i.e. all item fields from Zyte API), or if instead we should hard-code some short string (e.g. `(all fields)`) instead.

Reply:
Talked to @asadurski, agreed on keeping item entries when there are no overrides, and using a hard-coded `(all fields)` string as value.